ERA-11250 Swap in new cookie manager by Osana

chrisj-er opened the pull request:
### What does this PR do?
Swapping out OneTrust with Osana
feat: Osana cookie manager

### How does it look


### Relevant link(s)
* Tracking tickets: [ERA-\<ticket number\>](https://allenai.atlassian.net/browse/ERA-<ticket-number>)
* [Hosted demo environments](https://era-11250.pamdas.org)
* Etc

Commit:
ERA-11250 Swap in new cookie manager by Osana
feat: Osana cookie manager

Commit:
enable in production

Commit:
fix: update Osana check.

Commit:
fix: get spelling right for Osano

Commit:
optionally hide cookie settings menu item if Osano not fully initialized.

Commit:
fix: handle osano throwing an error of Language unavailable

Commit:
osano throwing promise rejection

Commit:
one more fix for changing locale in Osano

luixlive commented on src/UserMenu/index.js:
```diff
@@ -53,7 +53,11 @@ const UserMenu = ({ onLogOutClick, onProfileClick, selectedUserProfile, user, us
           <Dropdown.Divider />
         </>}
 
-        <Dropdown.Item onClick={() => cookieSettingsRef.current.click()}>{t('cookieSettingsItem')}</Dropdown.Item>
```
Seems like we can do a bit more cleanup in this file. The following lines can go away:
```
// Line 1 (just the useRef import)
import React, { useRef } from 'react';
...
// Line 17
const cookieSettingsRef = useRef();
...
// Line 28
<button className="ot-sdk-show-settings" hidden id="ot-sdk-btn" ref={cookieSettingsRef} />
```

luixlive commented on src/index.js:
```diff
@@ -62,9 +62,12 @@ const RootApp = () => {
 
   useEffect(() => {
-    if (window?.OneTrust) {
+    if (typeof window !== 'undefined' && window.Osano) {
```
We usually go for optional chaining, so we can keep the code as it was, just changing the property:
```
if (window?.Osano) {
```
It's logically the same.

JoshuaVulcan replied:
^^that. `window?.Osano?.cm` is a step safer as well, since it checks for the target object receiving a property in this code.

JoshuaVulcan commented on src/UserMenu/index.js:
```diff
@@ -53,7 +53,11 @@ const UserMenu = ({ onLogOutClick, onProfileClick, selectedUserProfile, user, us
         </>}
 
-        <Dropdown.Item onClick={() => cookieSettingsRef.current.click()}>{t('cookieSettingsItem')}</Dropdown.Item>
+        {window.Osano?.cm && (
```
Let's add some optional chaining to the window object (like `window?.Osano?.cm`) so we're not set up for future failures in headless test environments.

amicavi commented:
Before merging this ticket, please consider the impact this could have on [GDPR](https://docs.osano.com/hc/en-us/articles/22472073119764-Osano-and-GDPR)

Commit:
Merge branch 'develop' into ERA-11250

Commit:
Merge branch 'develop' into ERA-11250

Commit:
Merge branch 'develop' into ERA-11250

chrisj-er commented:
Osana is our new GDPR enabled Cookie Manager.

luixlive commented:
@chrisj-er this PR has been open for 8 months. There's a bit of feedback, and also an approval. Any context of when will we be ready to merge it (or close it in case we decide not to go with Osana)?

Commit:
Merge branch 'develop' into ERA-11250

Commit:
Removed useRef import
Replaced typeof window !== 'undefined' && window.Osano with window?.Osano?.cm. Removed the no-op locale fallback logic
Fixed indentation on the <% if %> block to align with surrounding content.
